### PR TITLE
rpc, wallet: add abandoned field for all categories of transaction in ListTransaction

### LINKED ISSUE #25130: gettransaction does not contain the field "abandoned" for abandoned receiving tx
In specter https://github.com/cryptoadvance/specter-desktop/issues/1491#issuecomment-1126145011  a user can abandon an **receiving** tx that was evicted from the mempool (for example because it was double spent). The tx is marked via [abandontransaction](https://github.com/bitcoin-dot-org/developer.bitcoin.org/blob/master/reference/rpc/abandontransaction.rst) as abandoned in bitcoin core.

However [gettransaction](https://github.com/bitcoin-dot-org/developer.bitcoin.org/blob/master/reference/rpc/gettransaction.rst) and [listtransactions](https://github.com/bitcoin-dot-org/developer.bitcoin.org/blob/master/reference/rpc/listtransactions.rst) do not return the field `abandoned`, even though this is visible in the GUI:

image

**Consequence**: Specter cannot get the `abandoned` information from bitcoin rpc, and will fetch the abandoned tx back again via [listtransactions](https://github.com/bitcoin-dot-org/developer.bitcoin.org/blob/master/reference/rpc/listtransactions.rst) after it was deleted in specter.   

**Describe the solution you'd like**
Always return the `abandoned` field in https://github.com/bitcoin/bitcoin/blob/225e5b57b2ee2bc1acd7f09c89ccccc15ef8c85f/src/wallet/rpc/transactions.cpp#L345  and not only in this special case https://github.com/bitcoin/bitcoin/blob/225e5b57b2ee2bc1acd7f09c89ccccc15ef8c85f/src/wallet/rpc/transactions.cpp#L375
That should then automatically solve  [gettransaction](https://github.com/bitcoin-dot-org/developer.bitcoin.org/blob/master/reference/rpc/gettransaction.rst)  via https://github.com/bitcoin/bitcoin/blob/225e5b57b2ee2bc1acd7f09c89ccccc15ef8c85f/src/wallet/rpc/transactions.cpp#L795

**Describe alternatives you've considered**
If specter can't get the information about abandoned tx from bitcoin, it would have to keep an internal list. That is possible, but redundant and can lead to many edge cases.

**Steps to reproduce**

- Wallet A:  Send 1 BTC to Wallet B  (rbf)
- Wallet A:  Replace tx with rbf to send all funds to wallet A
- Wallet B: Abandon tx (because no longer in mempool)
- Wallet B:  [gettransaction](https://github.com/bitcoin-dot-org/developer.bitcoin.org/blob/master/reference/rpc/gettransaction.rst) will not show the  `abandoned` field

### PR DESCRIPTION
**Pull request by brunoerg**
Fixes #25130

### CODE REVIEW

**Review by jonatack — COMMENTED** [review]
Quick readover of the code as a break from seminar readings, but a key point would be to consider why this field was not returned other than for sent txns (and if concept ACKs, ensure there is appropriate test coverage in the tests that assert on this field. Quick grep:)

```
test/functional/feature_backwards_compatibility.py:197:                        assert not(txs[1]["abandoned"])
test/functional/feature_backwards_compatibility.py:200:                        assert txs[3]["abandoned"]
test/functional/wallet_abandonconflict.py:197:            assert_equal(tx["abandoned"], False)
test/functional/wallet_abandonconflict.py:202:        assert_equal(double_spend["abandoned"], False)
test/functional/wallet_abandonconflict.py:131:            assert_equal(tx['abandoned'], True)
```

**Comment by brunoerg** [comment]
> a key point would be to consider why this field was not returned other than for sent txns

I couldn't find any resources about it, i'd like to understand more, btw Concept NACKs are welcome!

File: doc/release-notes.md [hunk 1]
```diff
@@ -83,6 +83,9 @@ Low-level changes
 RPC
 ---
 
+- The `gettransaction` and `listtransactions` RPCs now return
+  the `abandoned` field for all categories of transactions.
```
[the remark this answers is not in the document]
Comment (brunoerg) [reply doc/release-notes.md:87]: Yeah! Nice find, thanks!

File: src/wallet/rpc/transactions.cpp [hunk 2]
```diff
@@ -735,8 +734,7 @@ RPCHelpMan gettransaction()
                                 {RPCResult::Type::NUM, "vout", "the vout value"},
                                 {RPCResult::Type::STR_AMOUNT, "fee", /*optional=*/true, "The amount of the fee in " + CURRENCY_UNIT + ". This is negative and only available for the \n"
                                     "'send' category of transactions."},
-                                {RPCResult::Type::BOOL, "abandoned", /*optional=*/true, "'true' if the transaction has been abandoned (inputs are respendable). Only available for the \n"
-                                     "'send' category of transactions."},
+                                {RPCResult::Type::BOOL, "abandoned", /*optional=*/true, "'true' if the transaction has been abandoned (inputs are respendable)."},
```
[the remark this answers is not in the document]
Comment (brunoerg) [reply src/wallet/rpc/transactions.cpp:737]: I think so.

File: src/wallet/rpc/transactions.cpp [hunk 3]
```diff
@@ -482,8 +483,7 @@ RPCHelpMan listtransactions()
                         },
                         TransactionDescriptionString()),
                         {
-                            {RPCResult::Type::BOOL, "abandoned", /*optional=*/true, "'true' if the transaction has been abandoned (inputs are respendable). Only available for the \n"
-                                 "'send' category of transactions."},
+                            {RPCResult::Type::BOOL, "abandoned", "'true' if the transaction has been abandoned (inputs are respendable)."},
```
Comment (luke-jr) [inline src/wallet/rpc/transactions.cpp:486]: This description only makes any sense for sent transactions

**Comment by achow101** [comment]
Are you still working on this?

File: src/wallet/rpc/transactions.cpp (hunk 3 above)
Comment (brunoerg) [reply src/wallet/rpc/transactions.cpp:467]: Not specifically as we discussed on the issue.

**Comment by brunoerg** [comment]
> Are you still working on this?

Yes.

**Comment by achow101** [comment]
ACK efdb80a69000572a5f36fc702090da99c627238d

I think it's reasonable to output this for all transactions, even if we decide that one type cannot be abandoned. A trivial case where this is useful are send-to-self transactions as these show up as individual send and receive transactions.

**Comment by MarcoFalke** [comment]
You will have to modify the RPC docs for `gettransaction` as well

**Comment by brunoerg** [comment]
> You will have to modify the RPC docs for gettransaction as well

Nice find, thanks! Force-pushed modifying it.

**Comment by achow101** [comment]
re-ACK 0c520679ab5f0ba99584cb356ec28ef089f14735

**Pull request merged**